### PR TITLE
Feature/decode child object in parent object bunq/sdk php#68

### PR DESCRIPTION
**Pull request by OGKevin**
Closes bunq/sdk_php#68
Closes bunq/sdk_php#84

### CODE REVIEW

File: src/Model/Core/BunqModel.php [hunk 1]
```diff
@@ -121,17 +135,85 @@ protected static function createListFromResponseArray(
...
         if (is_null($responseArray)) {
             return null;
-        } else {
-            return self::createInstanceFromResponseArray($responseArray);
         }
+
+        return self::createInstanceFromResponseArray($responseArray);
+    }
+
+    /**
```
Comment (OGKevin) [inline src/Model/Core/BunqModel.php]: Array type

File: src/Model/Core/BunqModel.php [hunk 2]
```diff
@@ -121,17 +135,85 @@ protected static function createListFromResponseArray(
...
+        $model = self::createInstanceFromResponseArray($responseArray);
+
+        if ($model->isAllFieldNull() && $depthCounter < self::DEPTH_COUNTER_MAX) {
+            $model = self::decodeInsideModelFields($responseArray, $model, $depthCounter);
+        }
+
+        return $model;
+    }
+
+    /**
```
Comment (OGKevin) [inline src/Model/Core/BunqModel.php]: Array type

File: src/Model/Core/ServerPublicKey.php [hunk 3]
```diff
@@ -34,13 +34,29 @@ public function getServerPublicKey(): PublicKey
     }
 
     /**
-     * @param mixed[] $responseArray
+     * @param array $responseArray
```
Comment (OGKevin) [inline src/Model/Core/ServerPublicKey.php]: But why ? :sweat_smile:

File: src/Model/Core/BunqModel.php (hunk 1 above)
Comment (OGKevin) [reply src/Model/Core/BunqModel.php]: Done

File: src/Model/Core/BunqModel.php (hunk 2 above)
Comment (OGKevin) [reply src/Model/Core/BunqModel.php]: Done

File: src/Model/Core/BunqModel.php [hunk 4]
```diff
@@ -121,17 +135,85 @@ protected static function createListFromResponseArray(
...
+                    $depthCounter + self::DEPTH_COUNTER_INCREMENTER
+                );
+
+                if ($fieldContents->isAllFieldNull()) {
+                    $model->{$field} = null;
+                } else {
+                    $model->{$field} = $fieldContents;
+                }
+            } catch (BunqException $exception) {
+                continue;
```
Comment (andrederoos) [inline src/Model/Core/BunqModel.php]: mmm this sucks

File: src/Model/Core/BunqModel.php [hunk 5]
```diff
@@ -121,17 +135,85 @@ protected static function createListFromResponseArray(
...
+     * @param mixed[] $responseArray
+     * @param int|null $depthCounter
+     *
+     * @return BunqModel|null
+     */
+    private static function createFromResponseArrayAnchorObject(
+        array $responseArray,
+        int $depthCounter = null
+    ) {
+        if (is_null($responseArray)) {
```
Comment (andrederoos) [inline src/Model/Core/BunqModel.php]: empty?

File: src/Model/Core/BunqModel.php (hunk 4 above)
Comment (OGKevin) [reply src/Model/Core/BunqModel.php]: Correct I dont even remember why I did this 😅 let me find out.

**Comment by OGKevin** [comment]
@andrederoos All yours again.

**Pull request merged**